### PR TITLE
Check IsRecording in TelemetrySpan before adding event and attributes

### LINKED ISSUE #1163: Span/Tracer Shim should be consistent with OTel spec.
The shim should follow the spec as such. Some notable missing:
TelemetrySpan should have get for Context, parent.
TelemetrySpan should not contain getters/setter for SpanKind. etc.

All the methods to add attributes/events should be noops unless IsRecording = true.

Compare with this doc:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/947

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #1163.

## Changes

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Api/Trace/TelemetrySpan.cs [hunk 1]
```diff
@@ -380,5 +383,21 @@ protected virtual void Dispose(bool disposing)
...
+            {
+                this.Activity?.SetTag(key, value);
+            }
+        }
+
+        private void AddEventInternal(string name, DateTimeOffset timestamp = default, ActivityTagsCollection tags = null)
+        {
+            if (this.IsRecording)
+            {
+                this.Activity?.AddEvent(new ActivityEvent(name, timestamp, tags));
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:399]: null check not needed here as isrecording already does it.

Comment (eddynaka) [reply src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:399]: ok. i will remove in both places.

File: src/OpenTelemetry.Api/Trace/TelemetrySpan.cs [hunk 2]
```diff
@@ -301,7 +301,10 @@ public string GetBaggageItem(string key)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public TelemetrySpan AddBaggage(string key, string value)
         {
-            this.Activity?.AddBaggage(key, value);
+            if (this.IsRecording)
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:304]: its best to not touch this. as Mikel's PR is removing this. :)

Comment (eddynaka) [reply src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:304]: ok. will revert this change.

Comment (eddynaka) [reply src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:304]: done.

File: src/OpenTelemetry.Api/Trace/TelemetrySpan.cs [hunk 3]
```diff
@@ -380,5 +379,21 @@ protected virtual void Dispose(bool disposing)
         {
             this.Activity?.Dispose();
         }
+
+        private void SetAttributeInternal(string key, object value)
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/Trace/TelemetrySpan.cs:383]: Should we inline these?

**Pull request merged**